vernac/record.ml

```diff
@@ -291,26 +326,97 @@ let instantiate_possibly_recursive_type ind u ntypes paramdecls fields =
   let subst' = List.init ntypes (fun i -> mkIndU ((ind, ntypes - i - 1), u)) in
   Termops.substl_rel_context (subst @ subst') fields
 
-type projection_flags = {
-  pf_subclass: bool;
-  pf_canonical: bool;
-}
-
 (* We build projections *)
-let declare_projections indsp ctx ?(kind=Decls.StructureComponent) binder_name flags fieldimpls fields =
+
+(* TODO: refactor the declaration part here; this requires some
+   surgery as Evarutil.finalize is called too early in the path *)
+let build_named_proj ~primitive ~flags ~poly ~univs ~uinstance ~kind env paramdecls paramargs decl impls fid subst sp_projs nfi ti i indsp mib lifted_fields x rp =
```

Comment:
Can you write a comment documenting what this function does?

Reply:
Sure.

Reply:
It is pretty obvious tho, I'm unsure what to add, this builds the term of a projection `a : T`.